docs/shrinkwrap.md

```diff
@@ -3,22 +3,37 @@ Shrinkwrap
 We use `npm shrinkwrap --dev` to lock down our dependency versions. This allows us to
 freeze all dependencies at the exact version we have installed in our node_modules.
 
-See [shrinkwrap docs](https://docs.npmjs.com/cli/shrinkwrap)
+See: [shrinkwrap docs](https://docs.npmjs.com/cli/shrinkwrap)
 
 ## Modifying Dependencies
 
-- In the root directory delete local node_modules
+We use clingwrap to avoid creating huge diffs in npm-shrinkwrap.json. clingwrap also removes
+`from` and `resolved` fields which is expected. If you happen to be pointing at a package
+that is not published with npm, you may want to update the npm-shrinkwrap.json by hand.
+
+- Install [clingwrap](https://github.com/goodeggs/clingwrap) globally: `npm install -g clingwrap`
+- Update your desired package, e.g. `npm install lodash@4.0.0 --save`
+- Clingwrap your updated package `clingwrap lodash`
+- Verify that Calypso works as expected and that tests pass.
+- Commit the updated package.json and npm-shrinkwrap.json
+
+## Modifying Sub-Dependencies
+
+Periodically, we'll want to bump sub-dependencies to pick up bugfixes. This may result
+in a large diff that is too big to review. It's very important that your node_modules 
+is deleted before you do this to be sure to pick up the latest versions.
```

Comment:
> Testing instructions would be to make sure a clean install works, and that Calyso works.

OK, then I would say something like:

> This may result in a large diff that is too big to review, so testing instructions should ensure a clean install works and Calypso runs and tests correctly.

----

> To update sub-dependencies in a single module, ( with lodash as an example ) :

OK, but that's not quite what I'm saying. You're giving instructions to bump the sub-dependencies across modules, not in a single module. I'm just pointing out we might find an issue in a separate bumped sub-dependency. I think it's fine, but would be good to point out we're updating all modules and subdependencies when following the instructions in this list.